src/librustc_back/target/mod.rs

```diff
@@ -214,6 +214,7 @@ supported_targets! {
     ("le32-unknown-nacl", le32_unknown_nacl),
     ("asmjs-unknown-emscripten", asmjs_unknown_emscripten),
     ("wasm32-unknown-emscripten", wasm32_unknown_emscripten),
+    ("wasm32-unknown-unknown", wasm32_unknown_unknown),
```

Comment:
To add, WASM supports non-web target, and as a developer of an alternate wasm backend myself the emscripten-free target is ideal.